crates/burn-jit/src/kernel/pool/standard_pool2d_shader.rs

```diff
@@ -0,0 +1,243 @@
+        gpu!(scope, output_shape_2 = shape(output, 2u32));
+        gpu!(scope, output_shape_3 = shape(output, 3u32));
+
+        let pool_stride_0 = Variable::GlobalScalar(0, Elem::UInt);
+        let pool_stride_1 = Variable::GlobalScalar(1, Elem::UInt);
+        let dilation_0 = Variable::GlobalScalar(2, Elem::UInt);
+        let dilation_1 = Variable::GlobalScalar(3, Elem::UInt);
+        let padding_0 = Variable::GlobalScalar(4, Elem::UInt);
+        let padding_1 = Variable::GlobalScalar(5, Elem::UInt);
+
+        let b = scope.create_local(Elem::UInt);
+        let c = scope.create_local(Elem::UInt);
+        let oh = scope.create_local(Elem::UInt);
+        let ow = scope.create_local(Elem::UInt);
+
+        gpu!(scope, b = id / output_stride_0);
+        gpu!(scope, b = b % output_shape_0);
+
+        gpu!(scope, c = id / output_stride_1);
+        gpu!(scope, c = c % output_shape_1);
+
+        gpu!(scope, oh = id / output_stride_2);
+        gpu!(scope, oh = oh % output_shape_2);
+
+        gpu!(scope, ow = id / output_stride_3);
+        gpu!(scope, ow = ow % output_shape_3);
+
+        let ih = scope.create_local(Elem::UInt);
+        let iw = scope.create_local(Elem::UInt);
+        let dilated = scope.create_local(Elem::UInt);
+
+        let ih_pad = scope.create_local(Elem::UInt);
+        let iw_pad = scope.create_local(Elem::UInt);
+        let result = scope.create_local(input.item());
+
+        let index_input = scope.create_local(Elem::UInt);
+        let index_input_0 = scope.create_local(Elem::UInt);
+        let index_input_1 = scope.create_local(Elem::UInt);
+        let index_input_2 = scope.create_local(Elem::UInt);
+        let index_input_3 = scope.create_local(Elem::UInt);
+        let idx = scope.create_local(Elem::UInt);
+
+        let within_padding_h = scope.create_local(Elem::Bool);
+        let within_padding_w = scope.create_local(Elem::Bool);
+        let tmp_padding = scope.create_local(Elem::Bool);
+        let border_bottom = scope.create_local(Elem::UInt);
+        let border_right = scope.create_local(Elem::UInt);
+
+        gpu!(scope, border_bottom = input_shape_2 + padding_0);
+        gpu!(scope, border_right = input_shape_3 + padding_1);
+
+        gpu!(scope, index_input_0 = b * input_stride_0);
+        gpu!(scope, index_input_1 = c * input_stride_1);
+
+        let accumulator = self.pool_strategy.initialize(scope, input.item());
+
+        (0..self.kernel_size[0]).for_each(|kh| {
+            gpu!(scope, ih = oh * pool_stride_0);
+            gpu!(scope, dilated = kh * dilation_0);
+            gpu!(scope, ih += dilated);
+
+            gpu!(scope, within_padding_h = ih >= padding_0);
+            gpu!(scope, tmp_padding = ih < border_bottom);
+            gpu!(scope, within_padding_h = within_padding_h && tmp_padding);
+
+            gpu!(scope, if (within_padding_h).then(|scope| {
+                (0..self.kernel_size[1]).for_each(|kw| {
```

Comment:
A good optimization would be to have a propertie that checks if we need to perform those bound checks (similar to matmul) Can be in another PR.